src/libraries/System.Formats.Tar/tests/TarEntry/GnuTarEntry.Tests.cs

```diff
@@ -93,5 +94,469 @@ public void SupportedEntryType_Fifo()
             SetFifo(fifo);
             VerifyFifo(fifo);
         }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void DataOffset_RegularFile(bool canSeek)
+        {
+            byte expectedData = 5;
+            using MemoryStream ms = new();
+            using (TarWriter writer = new(ms, leaveOpen: true))
+            {
+                GnuTarEntry entry = new GnuTarEntry(TarEntryType.RegularFile, InitialEntryName);
+                entry.DataStream = new MemoryStream();
+                entry.DataStream.WriteByte(5);
```

Comment:
```suggestion
                entry.DataStream.WriteByte(expectedData);
```

A few instances of this.  Consider just replacing all uses of `5` with some shared constant.